element-connector/src/main/java/org/eclipse/californium/elements/util/ExecutorsUtil.java

```diff
@@ -169,21 +226,42 @@ public static void shutdownExecutorGracefully(long timeMaxToWaitInMs, ExecutorSe
 				executor.shutdownNow();
 			}
 			Thread.currentThread().interrupt();
+		} finally {
+			time = TimeUnit.NANOSECONDS.toMillis(ClockUtil.nanoRealtime() - start);
+			if (time > timeMaxToWaitInMs) {
+				LOGGER.warn("await termination {} ms exceeded the maximum {} ms", time, timeMaxToWaitInMs);
+			}
+		}
+	}
+
+	/**
+	 * Set remove on cancel policy of provided executor.
+	 * 
+	 * Ignored, if not supported by java vm or executors implementation.
+	 * 
+	 * @param executor executor to set remove on cancel policy
+	 * @since 3.0
+	 */
+	@NotForAndroid
+	private static void setRemoveOnCancelPolicy(ScheduledExecutorService executor) {
+		if (REMOVE_ON_CANCEL != null && executor instanceof ScheduledThreadPoolExecutor) {
+			((ScheduledThreadPoolExecutor) executor).setRemoveOnCancelPolicy(REMOVE_ON_CANCEL);
 		}
 	}
 
 	/**
-	 * Executor, which uses a {@link ScheduledThreadPoolExecutor}
-	 * with {@link ExecutorsUtil#SPLIT_THRESHOLD} threads for scheduling, and an
+	 * Executor, which uses a {@link ScheduledThreadPoolExecutor} with
+	 * {@link ExecutorsUtil#SPLIT_THRESHOLD} threads for scheduling, and an
 	 * additional {@link ThreadPoolExecutor} for execution. This may result in
 	 * better performance for direct job, when many scheduled job are queued for
 	 * execution.
 	 * 
-	 * Note: IT'S A WORKAROUND! IT MAY BE REPLACED IN THE FUTURE! See issue #690.
+	 * Note: IT'S A WORKAROUND! IT MAY BE REPLACED IN THE FUTURE! See issue
+	 * #690.
 	 */
 	private static class SplitScheduledThreadPoolExecutor extends ScheduledThreadPoolExecutor {
 
-		private static final long QUEUE_SIZE_DIFF = 20000;
+		private final long SCHEDULE_EXECUTOR_LOGGING_QUEUE_SIZE_DIFF_DEFAULT = 10000;
```

Comment:
Reading this quiclky, I see this change which is maybe not intended ?

Reply:
It's only for logging, and may now be adapted/changed by EXECUTER_LOGGING_QUEUE_SIZE_DIFF.
That logging helps to see, when the jobs are actually removed. In difference to the web, that logging indicated in my tests, that the postpone of removing jobs is small. At least, if there are always new jobs.